pkg/apis/cr/v1alpha1/types.go

```diff
@@ -249,12 +249,14 @@ type CredentialType string
 
 const (
 	CredentialTypeKeyPair CredentialType = "keyPair"
+	CredentialTypeAws     CredentialType = "aws"
```

Comment:
Rather than include the type `Aws` here, let's make this more general purpose since the type will already be captured by the secret.